src/core/services.py

```
@@ -123,13 +128,29 @@ def run(
             for i, row in enumerate(participants_df.to_dict("records"))
         ]
 
+        hints = None
+        if previous_results is not None and not previous_results.empty:
+            if (
+                config.COL_GROUP in previous_results.columns
+                and "_original_index" in previous_results.columns
+            ):
```

Ruff (0.15.12)

[warning] Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)